app/services/feeds/import.rb

```diff
@@ -0,0 +1,192 @@
+
+        [user_id, parsed_feed]
+      rescue StandardError => e
+        # TODO: add better exception handling (eg. rescueing Feedjira::NoParserAvailable separately)
+        report_error(
+          e,
+          feeds_import_info: {
+            user_id: user_id,
+            error: "Feeds::Import::ParseFeedError"
+          },
+        )
+
+        next
+      end
+
+      result.compact.to_h
+    end
+
+    # TODO: currently this is exactly as in RSSReader, but we might find
+    # avenues for optimization, like:
+    # 1. why are we sending N exists query to the DB, one per each item, can we fetch them all?
+    # 2. should we queue a batch of workers to create articles, but then, following issues ensue:
+    # => synchronization on write (table/row locking)
+    # => what happens if 2 jobs are in the queue for the same article?
+    # => what happens if they stay in the queue for long and the next iteration of the feeds importer starts?
+    def create_articles_from_user_feed(user, feed)
+      articles = []
+
+      feed.entries.reverse_each do |item|
+        next if medium_reply?(item) || article_exists?(user, item)
+
+        feed_source_url = item.url.strip.split("?source=")[0]
+        article = Article.create!(
+          feed_source_url: feed_source_url,
+          user_id: user.id,
+          published_from_feed: true,
+          show_comments: true,
+          body_markdown: RssReader::Assembler.call(item, user, feed, feed_source_url),
+          organization_id: nil,
+        )
+
+        articles.append(article)
+      rescue StandardError => e
+        # TODO: add better exception handling
+        report_error(
+          e,
+          feeds_import_info: {
+            username: user.username,
+            feed_url: user.feed_url,
+            item_count: get_item_count_error(feed),
+            error: "Feeds::Import::CreateArticleError:#{item.url}"
+          },
+        )
+
+        next
+      end
+
+      articles
+    end
+
+    def get_host_without_www(url)
+      url = "http://#{url}" if URI.parse(url).scheme.nil?
+      host = URI.parse(url).host.downcase
+      host.start_with?("www.") ? host[4..] : host
+    end
+
+    def medium_reply?(item)
+      get_host_without_www(item.url.strip) == "medium.com" &&
+        !item[:categories] &&
+        content_is_not_the_title?(item)
+    end
+
+    def content_is_not_the_title?(item)
+      # [[:space:]] removes all whitespace, including unicode ones.
+      content = item.content.gsub(/[[:space:]]/, " ")
+      title = item.title.delete("…")
+      content.include?(title)
+    end
+
+    def article_exists?(user, item)
+      title = item.title.strip.gsub('"', '\"')
+      feed_source_url = item.url.strip.split("?source=")[0]
+      relation = user.articles
+      relation.where(title: title).or(relation.where(feed_source_url: feed_source_url)).exists?
+    end
+
+    def report_error(error, metadata)
+      Honeybadger.context(metadata)
```

Comment:
We get a lot of Honeybadgers right now from the RSS reader but its relatively quiet bc they are all repeats now instead of new. This is likely going to be super noisy, what if we logged these instead so we dont sound any false alarms in HB?

Reply:
As I can't see the logs in production I'm not sure what would be the best decision here. Let's keep in mind that this is mostly to be run 2 o 3 times to have a general idea of speed and memory occupation. I don't consider it the final version for all the conversations we had in the issue thread

Reply:
In a perfect situation we should only log errors that are actual errors. Like a feed being removed upstream shouldn't be an error, we should just stop fetching it eventually. Also: timeout errors are not super useful in Honeybadger as they might depend on the network conditions. In a sense, I need to know which errors can occurr in production, to know which ones to filter out :)

Reply:
> As I can't see the logs in production I'm not sure what would be the best decision here. 

You should have access to all the [error logs in Datadog](https://app.datadoghq.com/logs?cols=host%2Cservice%2C%40syslog.procid&from_ts=1603992686819&index=&live=true&messageDisplay=inline&stream_sort=desc&to_ts=1603993586819) which is why I am suggesting using them. Let me know if you don't and I will investigate what the problem might be. 

> Let's keep in mind that this is mostly to be run 2 o 3 times to have a general idea of speed and memory occupation. 
Right, those few times will set off a lot of new HB which will be noisy

> In a perfect situation, we should only log errors that are actual errors.
100% agree, but if we log info statements and we want to see them then I have to turn on INFO logging which will create a bunch of logs we dont need. I am merely suggesting using the error logs as a temporary solution here to give us the insight we need into the process while also not interfering with our alerting system and causing alert fatigue.

Reply:
Ah got it, thanks. For some reason I still thought logs were only going to Heroku's drain. I'll switch them to error logs instead of HB reports